auth.php

```diff
@@ -500,6 +494,11 @@ public function saml_login() {
         $passive = $this->config->duallogin == saml2_settings::OPTION_DUAL_LOGIN_PASSIVE;
         $passive = (bool)optional_param('passive', $passive, PARAM_BOOL);
         $params = ['isPassive' => $passive];
+        // When we return from authenticating, we also return the idp with a url param
+        // This will be found using $_GET['idp'] above
+        $md5idpentityid = auth_saml2_get_md5_idpentityid();
+        $params = ['ReturnTo' => $ME . '?idp=' . $md5idpentityid];
```

Comment:
I don't think this will work if $ME has ?foo=bar in it. Please use the moodle_url object to add a param